### PR TITLE
Replace paper-listbox in cast frontend

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Replace paper-listbox in cast frontend. <s>Please check visually before approval since I was not able to test this fully. </s>

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **Refactor**
  - Improved list items structure in the `HcCast` class for better performance and maintainability.
  - Updated event handling for picking views to enhance user interaction.

- **Style**
  - Updated styles related to list items for a more consistent and modern appearance.

### CODE REVIEW

File: cast/src/launcher/layout/hc-cast.ts [hunk 1]
```diff
@@ -83,34 +83,33 @@ class HcCast extends LitElement {
               `
             : html`
                 <div class="section-header">PICK A VIEW</div>
-                <paper-listbox
-                  attr-for-selected="data-path"
-                  .selected=${this.castManager.status.lovelacePath || ""}
-                >
+                <mwc-list>
```
Comment (bramkragten) [inline cast/src/launcher/layout/hc-cast.ts:86]: Shouldn't this have a `value`?

File: cast/src/launcher/layout/hc-cast.ts [hunk 2]
```diff
@@ -83,34 +83,33 @@ class HcCast extends LitElement {
...
                       generateDefaultViewConfig({}, {}, {}, {}, () => ""),
                     ]
                   ).map(
-                    (view, idx) => html`
-                      <paper-icon-item
+                    (view, idx) =>
+                      html`<ha-list-item
+                        graphic="avatar"
                         @click=${this._handlePickView}
                         data-path=${view.path || idx}
```
Comment (bramkragten) [inline cast/src/launcher/layout/hc-cast.ts:96]:
```suggestion
                        .value=${view.path || idx}
```

File: cast/src/launcher/layout/hc-cast.ts [hunk 3]
```diff
@@ -83,34 +83,33 @@ class HcCast extends LitElement {
...
+                        selected=${this.castManager.status?.lovelacePath || ""}
```
Comment (bramkragten) [inline cast/src/launcher/layout/hc-cast.ts:97]:
```suggestion
```

File: cast/src/launcher/layout/hc-cast.ts [hunk 4]
```diff
@@ -83,34 +83,33 @@ class HcCast extends LitElement {
...
                     this.lovelaceViews ?? [
                       generateDefaultViewConfig({}, {}, {}, {}, () => ""),
                     ]
                   ).map(
-                    (view, idx) => html`
-                      <paper-icon-item
+                    (view, idx) =>
+                      html`<ha-list-item
+                        graphic="avatar"
                         @click=${this._handlePickView}
```
Comment (bramkragten) [inline cast/src/launcher/layout/hc-cast.ts:95]: We should not use a click handler here, as it should also work when navigating by keyboard.

**Comment by silamon** [comment]
Monthly revisit: I'm going to keep this in draft until #19955 passes through since both PR's may receive the same comments.

**Review by bramkragten — DISMISSED** [review]

File: cast/src/launcher/layout/hc-cast.ts [hunk 5]
```diff
@@ -83,34 +83,34 @@ class HcCast extends LitElement {
               `
             : html`
                 <div class="section-header">PICK A VIEW</div>
-                <paper-listbox
-                  attr-for-selected="data-path"
-                  .selected=${this.castManager.status.lovelacePath || ""}
-                >
+                <mwc-list @action=${this._handlePickView} activatable>
```
[the remark this answers is not in the document]
Comment (silamon) [reply cast/src/launcher/layout/hc-cast.ts:86]: That doesn't work until you have a selectable list (list item with checkboxes): https://www.npmjs.com/package/@material/mwc-list#mwc-list-2

**Review by bramkragten — APPROVED** [review]

**Pull request merged**